pkg/gadgets/audit-seccomp/tracer/bpf/audit-seccomp.c

```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: GPL-2.0
+/* Copyright (c) 2021 The Inspektor Gadget authors */
+
+/* This BPF program uses the GPL-restricted function bpf_probe_read*().
+ */
+
+#include <vmlinux/vmlinux.h>
+
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_core_read.h>
+#include <bpf/bpf_tracing.h>
+
+#include "audit-seccomp.h"
+
+#include <gadgettracermanager/bpf-maps.h>
+
+/* The stack is limited, so use a map to build the event */
+struct {
+	__uint(type, BPF_MAP_TYPE_PERCPU_ARRAY);
+	__uint(max_entries, 1);
+	__type(key, __u32);
+	__type(value, struct event);
+} tmp_event SEC(".maps");
+
+struct {
+	__uint(type, BPF_MAP_TYPE_PERF_EVENT_ARRAY);
+} events SEC(".maps");
+
+SEC("kprobe/audit_seccomp")
+int kprobe__audit_seccomp(struct pt_regs *ctx)
+{
+	unsigned long syscall = PT_REGS_PARM1(ctx);
+	int code = PT_REGS_PARM3(ctx);
+
+	struct task_struct *task = (struct task_struct *)bpf_get_current_task();
+	__u64 mntns_id = BPF_CORE_READ(task, nsproxy, mnt_ns, ns.inum);
+	if (mntns_id == 0) {
+		return 0;
+	}
+
+#ifdef WITH_FILTER
```

Comment:
For this, I think we prefer to use a `const volatile bool filter_by_mntns` and using `spec.RewriteConstants` or is it something totally different?

Reply:
For this specific `#ifdef`, we could do with a `const volatile bool` parameter instead.

But this was done like this in the process-collector gadget to have 2 versions of the BPF program with and without the BPF map filter, see:
https://github.com/kinvolk/inspektor-gadget/blob/ff7a7b162643bda344f7750f6e369c4988051873/pkg/gadgettracermanager/bpf-maps.h#L8

Because we cannot control the presence of this BPF map with a `const volatile bool` parameter.

This Makefile controls it (`-DWITH_FILTER`):
https://github.com/kinvolk/inspektor-gadget/blob/ff7a7b162643bda344f7750f6e369c4988051873/pkg/gadgets/process-collector/tracer/bpf/Makefile#L6-L10

See previous discussion: https://github.com/kinvolk/inspektor-gadget/pull/189#discussion_r677495432
Do you think we should revisit it and use a dummy map instead of the `ifdef`?

For the audit-seccomp gadget, I didn't implement the filtering yet, so the `#ifdef WITH_FILTER` is currently always false (TODO).